projectq/backends/_azure/_azure_quantum_test.py

```diff
@@ -0,0 +1,645 @@
+# -*- coding: utf-8 -*-
+#   Copyright 2021 ProjectQ-Framework (www.projectq.ch)
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from unittest import mock
+
+import projectq.backends._azure._azure_quantum
+from projectq.ops import H, CX, All, Measure
+from projectq.cengines import MainEngine, BasicMapperEngine
+from projectq.backends import AzureQuantumBackend
+from projectq.backends._azure._exceptions import AzureQuantumTargetNotFoundError
+
+from azure.quantum import Workspace
```

Comment:
Make sure all the tests below are properly skipped when `azure_quantum` is not installed on the user's system.

Reply:
This is kind of unclear to me! 

If we skip module from testing because its dependency is not installed, it will hide any failing scenarios dependent to that module. Isn't it better to enforce user (or same applies in ci pipeline) to run all test cases for 100% coverage?

User can always use targeted director testing (or) `--ignore` argument explicitly if required. 

Let me know your thoughts.